packages/salesforce-adapter/src/filters/history_tracking.ts

```diff
@@ -0,0 +1,314 @@
+       * 'after' part of these changes based on the field changes we receive here.
+       *
+       * Note that we can't just use the changes from preDeploy as-is because some of them could be missing from the
+       * onDeploy changes because they failed to deploy, and we shouldn't re-add them to the list of changes.
+       * */
+      const createAdditionalObjectTypeChanges = async (
+        fieldChanges: ModificationChange<Field>[],
+        typesToIgnore: Set<string>
+      ): Promise<Change<ObjectType>[]> => {
+        const additionalChanges: Record<string, Change<ObjectType>> = {}
+
+        const addTrackedFieldChangeToObjectTypeChange = async (change: ModificationChange<Field>): Promise<void> => {
+          const [before, after] = getAllChangeData(change)
+          const parentTypeName = await apiName(after.parent)
+
+          const trackedBefore = before.annotations[FIELD_ANNOTATIONS.TRACK_HISTORY]
+          const trackedAfter = after.annotations[FIELD_ANNOTATIONS.TRACK_HISTORY]
+
+          if (!!trackedBefore === !!trackedAfter) {
+            // converting to bool because some types don't have the annotation at all, but for now we still add a false
+            // annotation in preDeploy
+            return
+          }
+
+          if (!(parentTypeName in additionalChanges)) {
+            const originalObjectType = objectTypesChangedInPreDeploy[parentTypeName]
+            additionalChanges[parentTypeName] = toChange(
+              { before: originalObjectType, after: originalObjectType.clone() }
+            )
+          }
+
+          const changedObjectType = getChangeData(additionalChanges[parentTypeName])
+          if (trackedAfter) {
+            // field is added to tracked fields
+            changedObjectType.annotations[HISTORY_TRACKED_FIELDS][after.name] = new ReferenceExpression(after.elemID)
+          } else {
+            // field is removed from tracked fields
+            _.omit(changedObjectType.annotations[HISTORY_TRACKED_FIELDS], [after.name])
+          }
+        }
+
+        await awu(fieldChanges)
+          .filter(isModificationChange)
+          .filter(change => _.isEqual(_.omit(change.data.before.annotations[HISTORY_TRACKED_FIELDS]),
+            _.omit(change.data.after.annotations[HISTORY_TRACKED_FIELDS])))
+          .filter(async change => !typesToIgnore.has(await apiName(getChangeData(change).parent)))
+          .forEach(change => addTrackedFieldChangeToObjectTypeChange(change))
+
+        return Object.values(additionalChanges)
+      }
+
+      const changedCustomObjects = await awu(changes)
+        .filter(isAdditionOrModificationChange)
+        .filter(isObjectTypeChange)
+        .map(getChangeData)
+        .filter(isCustomObject)
+        .toArray()
+
+      changedCustomObjects
+        .forEach(centralizeHistoryTrackingAnnotations)
+
+      const changedCustomObjectNames = new Set(await awu(changedCustomObjects)
+        .map(objType => apiName(objType))
+        .toArray())
```

Comment:
readability nit (hard to keep track of the brackets like this and harder to understand what the `.map` operates on)
```suggestion
      const changedCustomObjectNames = new Set(
        await awu(changedCustomObjects)
          .map(objType => apiName(objType))
          .toArray()
      )
```

Reply:
Ack